### PR TITLE
UC: add new node

### PR DESCRIPTION
**Pull request by nikipuk**
## Related Issues

- Closes #79 

## Definition of Done

- [x] User Story
  - [x] All Acceptance Criteria met
- [x] Code
  - [x] No more code needed
  - [x] No known bugs
- [x] Clean Code
  - [ ] Comments have been created for new code
  - [x] ESLinter found no issues (editor & executor only)
  - [x] No unavoidable Code Smells
- [x] Testing
  - [x] Existing Tests pass
  - [x] New Unit Tests created
  - [x] New Unit Tests pass
- [x] SonarCloud Quality Gate passed

## Definition of more than Done (optional)

- [ ] Testing
  - [ ] New Integration Tests created
  - [ ] New Integration Tests passed

### CODE REVIEW

File: src/editor/components/flow/new-node-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
+import { Dialog } from '../dialog'
+import { Button } from '../form/button'
+import { Node as ReactFlowNode } from 'react-flow-renderer/nocss'
+
+interface AddNodeDialogProps {
+  isAddNodeOpen: boolean,
+
```
Comment (MrF3lix) [inline src/editor/components/flow/new-node-dialog.tsx:7]: Remove empty lines

File: src/editor/components/flow/new-node-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  flow,
+  nodes,
+  setNodes
+}: AddNodeDialogProps ): JSX.Element => {
+
+  const randomIntFromInterval = ( min: number, max: number ): number => {
+    return Math.floor( Math.random() * ( max - min + 1 ) + min )
+  }
+
+  const newHttpInNode = (): ReactFlowNode => {
```
Comment (MrF3lix) [inline src/editor/components/flow/new-node-dialog.tsx:32]: Can we make this less repetitive? Maybe by using a class for each node so you can just use 'new HttpInputNode()'

File: src/editor/components/flow/new-node-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  }
+
+  return (
+    <Dialog
+      isOpen={isAddNodeOpen}
+      onClose={(): void => setIsAddNodeOpen( false )}
+      title={`Add Node to: ${flow.name}`}
+    >
+      <div className="py-4 flex flex-col gap-3 w-px-{400} dark:text-gray-100">
+        <h4>To be changed</h4>
```
Comment (MrF3lix) [inline src/editor/components/flow/new-node-dialog.tsx:105]: You can add the correct title here :)

File: src/editor/pages/flows/[id].tsx [hunk 4]
```diff
@@ -81,30 +95,28 @@ const DataFlowOverview = ( { dataFlow }: any ): JSX.Element => {
       method: 'DELETE',
       headers: { 'Content-Type': 'application/json' },
       body: JSON.stringify( flow )
-    }  )
-    if( response.ok ) {
+    } )
+    if ( response.ok ) {
```
Comment (MrF3lix) [inline src/editor/pages/flows/[id].tsx:99]: Is this linted?

**Review by MrF3lix — CHANGES_REQUESTED** [review]

File: src/editor/components/flow/new-node-dialog.tsx (hunk 2 above)
Comment (nikipuk) [reply src/editor/components/flow/new-node-dialog.tsx:33]: yea I thought about that. I'll make some classes to clear up the new-node-dialog component

File: src/editor/components/flow/new-node-dialog.tsx (hunk 3 above)
Comment (nikipuk) [reply src/editor/components/flow/new-node-dialog.tsx]: do you mean on line 102 ?

File: src/editor/pages/flows/[id].tsx (hunk 4 above)
Comment (nikipuk) [reply src/editor/pages/flows/[id].tsx:99]: yes, but my linter is still not working properly 😟

File: src/editor/components/flow/new-node-dialog.tsx (hunk 3 above)
Comment (MrF3lix) [reply src/editor/components/flow/new-node-dialog.tsx]: 105 but seems to be fixed now.

File: src/editor/components/flow/new-node-dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    <Dialog
+      isOpen={isAddNodeOpen}
+      onClose={(): void => setIsAddNodeOpen( false )}
+      title={`Add Node to: ${flow.name}`}
+    >
+      <hr />
+      <div className="grid gap-4 grid-cols-1 sm:grid-cols-2 py-4 dark:text-gray-100">
+        <NodeTypeOption title="HTTP Input" description="A HTTP Input Node, used to get data from a http request. This node can be used as the start Node of your flow." onClick={(): void => addNode( new HttpInNode() )}/>
+        <NodeTypeOption title="HTTP Output" description="A HTTP Output Node, used to send data in the form of a http request. This node can be used as the end Node of your flow." onClick={(): void => addNode( new HttpOutNode() )}/>
+        <NodeTypeOption title="Filter" description="A Filter Node, useful for filtering the data flowing through your data flow. Don't need parts of your incoming data? Just add a Filter Node!" onClick={(): void => addNode( new FilterNode() )}/>
```
Comment (MrF3lix) [inline src/editor/components/flow/new-node-dialog.tsx:40]: Can you add this in a separate array of node types?

File: src/editor/components/graph/node-types.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+class Node implements ReactFlowNode {
+  id: string
+  dragHandle: string
+  type: string
+  data: any
+  position: XYPosition
+
+  constructor( type: string, data: any ) {
+    this.id = 'id-' + Date.now()
```
Comment (MrF3lix) [inline src/editor/components/graph/node-types.ts:12]: Good for now but we might want to use something else here? Do we still have the uuid package?

**Review by MrF3lix — CHANGES_REQUESTED** [review]

File: src/editor/components/graph/node-types.ts (hunk 6 above)
Comment (nikipuk) [reply src/editor/components/graph/node-types.ts:12]: good idea

File: src/editor/components/flow/new-node-dialog.tsx (hunk 5 above)
Comment (nikipuk) [reply src/editor/components/flow/new-node-dialog.tsx]: I sure can

**Review by MrF3lix — APPROVED** [review]

**Pull request merged**